components/types.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2023 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { createViewState as createCircularViewState } from "@jbrowse/react-circular-genome-view";
+import { createViewState as createLinearViewState } from "@jbrowse/react-linear-genome-view";
+
+export type JbrowseFileInputFormats = "BAM" | "VCF";
+
+export type JbrowseFileInputInfo = {
+  [k in JbrowseFileInputFormats]: string;
+};
+
+export type CircularViewModel = ReturnType<typeof createCircularViewState>;
+export type LinearViewModel = ReturnType<typeof createLinearViewState>;
```

Comment:
Discussion not request:
Would it be simpler to export types from the same file as we are exporting the functions?

If you adopt the barrel pattern for exports in all your directories it makes it much simpler to maintain all the type and const exports when the types live beside their definitions.